src/liiteri/api.clj

```diff
@@ -26,41 +27,30 @@
         :middleware [upload/wrap-multipart-params]
         :return schema/File
         (try
-          (response/ok (s3-store/create-file file s3-client db))
+          (response/ok (file-store/create-file file storage-engine db))
           (finally
-            (.delete (:tempfile file)))))
+            (io/delete-file (:tempfile file) true))))
 
       (api/GET "/files/metadata" []
         :summary "Get metadata for one or more files"
         :query-params [key :- (api/describe [s/Str] "Key of the file")]
         :return [schema/File]
-        (let [metadata (file-store/get-metadata key db)]
+        (let [metadata (file-metadata-store/get-metadata key db)]
           (if (> (count metadata) 0)
             (response/ok metadata)
```

Comment:
Should we just assume there's only one file and return one metadata here? I.e. return `schema/File` instead of `[schema/File]`. That would be easier for the users of this API.